### DISCUSSION TITLE
How to create special characters image with vips_text?

### DISCUSSION THREAD

**Discussion by rebornfz** [libvips/libvips#3325] [Q&A] [answered]
I need to create text image with special characters.
I put markup in strings with HTML , like this:
```
vips::VImage text_image = vips::VImage::text("<span font_desc=\"Roboto Regular\" letter_spacing=\"0\" size=\"34816\" foreground=\"#000000\">&amp;74.00</span>",
                vips::VImage::option ()->
                set("align",  VIPS_ALIGN_LOW) ->
                set("rgba", 1 ));
```
I want to create the following special characters

![image](https://user-images.githubusercontent.com/27042311/216816522-f4b4641a-c27a-4b1e-998b-ffea7693aa8f.png)

But only these worked well:
![image](https://user-images.githubusercontent.com/27042311/216816625-e98bed63-7adb-4187-802a-743dc13109f4.png)
![text](https://user-images.githubusercontent.com/27042311/216816864-bf19cd2e-3578-4908-a34d-a5ff10337446.png)

Others get the error:
```
terminate called after throwing an instance of 'vips::VError'
  what():  text: invalid markup in text
```

**Comment by jcupitt** [comment, marked as answer]
Hi again, `vips_text()` accepts pango markup, which is only a subset of HTML:

https://docs.gtk.org/Pango/pango_markup.html

But the string is utf-8, so you can just insert special characters in that form.
